libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -22,127 +62,134 @@ def store() -> SQLServer_VectorStore:
-
     # Length of ids returned by add_texts function should be equal to length of texts.
     assert len(result) == len(texts)
 
 
-def test_add_document_with_sqlserver(store: SQLServer_VectorStore) -> None:
+def test_add_document_with_sqlserver(
+    store: SQLServer_VectorStore,
+    docs: List[Document],
+) -> None:
     """Test that when add_document function is used, it integrates well
     with the add_text function in SQLServer Vector Store."""
-    docs = [
-        Document(
-            page_content="rabbit",
-            metadata={"color": "black", "type": "pet", "length": 6},
-        ),
-        Document(
-            page_content="cherry",
-            metadata={"color": "red", "type": "fruit", "length": 6},
-        ),
-        Document(
-            page_content="hamster",
-            metadata={"color": "brown", "type": "pet", "length": 7},
-        ),
-        Document(
-            page_content="cat", metadata={"color": "black", "type": "pet", "length": 3}
-        ),
-        Document(
-            page_content="elderberry",
-            metadata={"color": "blue", "type": "fruit", "length": 10},
-        ),
-    ]
     result = store.add_documents(docs)
     assert len(result) == len(docs)
 
 
 def test_that_a_document_entry_without_metadata_will_be_added_to_vectorstore(
     store: SQLServer_VectorStore,
+    docs: List[Document],
 ) -> None:
-    docs = [
-        Document(
-            page_content="rabbit",
-            metadata={"color": "black", "type": "pet", "length": 6},
-        ),
-        Document(
-            page_content="cherry",
-        ),
-        Document(
-            page_content="hamster",
-            metadata={"color": "brown", "type": "pet", "length": 7},
-        ),
-        Document(page_content="cat"),
-        Document(
-            page_content="elderberry",
-            metadata={"color": "blue", "type": "fruit", "length": 10},
-        ),
-    ]
-    result = store.add_documents(docs)
-    assert len(result) == len(docs)
+    """Test that you can add a document that has no metadata into the vectorstore."""
+    documents = docs[:-1]
+    documents.append(Document(page_content="elderberry"))
+    result = store.add_documents(documents)
+    assert len(result) == len(documents)
+
+
+def test_that_drop_deletes_vector_store(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Test that when drop is called, vector store is deleted
+    and a call to add_text raises an exception.
+    """
+    store.drop()
+    with pytest.raises(DBAPIError):
+        store.add_texts(texts)
+
+
+def test_that_similarity_search_returns_expected_no_of_documents(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Test that the amount of documents returned when similarity search
+    is called is the same as the number of documents requested."""
+    store.add_texts(texts)
+    result = store.similarity_search(query="Good review", k=3)
+    assert len(result) == 3
```

Comment:
How do we know expected number of docs is 3?

Reply:
3 is the value of k. I'll define a variable for it outside of the function call.